core-android/src/main/java/com/uber/sdk/android/core/auth/LoginManager.java

```diff
@@ -363,4 +365,26 @@ private void handleResultOk(@Nullable Intent data) {
 
         }
     }
+
+    private void validateRedirectUriRegistration(@NonNull Activity activity) {
+
+        String generatedRedirectUri = activity.getPackageName().concat(".uberauth://redirect");
+        String setRedirectUri = sessionConfiguration.getRedirectUri();
+        if (setRedirectUri != null &&
+                !generatedRedirectUri.equals(setRedirectUri)
+                && !AuthUtils.isRedirectUriRegistered(activity, Uri.parse(setRedirectUri))) {
+            String error = "Misconfigured redirect_uri. See https://github.com/uber/rides-android-sdk#authentication-migration-version-08-and-above "
+                    + "for more info. Either 1) Register " + generatedRedirectUri + " as a "
+                    + "redirect uri for the app at https://developer.uber.com/dashboard/ and "
+                    + "specify this in your SessionConfiguration or 2) Override the default "
+                    + "redirect_uri with the current one set (" + setRedirectUri + ") in the "
+                    + "AndroidManifest.";
+
+            if (Utility.isDebugable(activity)) {
+                throw new IllegalStateException(error);
```

Comment:
I think I'm missing the point of this, why do we throw an exception if the app is Debugable?

Reply:
To force folks to upgrade their implementation, otherwise it'll be broken (or in release, fall back to the deprecated webview)